gui/wxpython/dbmgr/base.py

```diff
@@ -287,12 +287,24 @@ def LoadData(self, layer, columns=None, where=None, sql=None):
 
             record = record.split(fs)
             if len(columns) != len(record):
-                GError(parent=self,
-                       message=_("Inconsistent number of columns "
-                                 "in the table <%(table)s>.") %
-                       {'table': tableName})
+                # Assuming there will be always at least one.
+                last = record[-1]
+                show_max = 3
+                if len(record) > show_max:
+                    record = record[:show_max]
+                # TODO: The real fix here is to use JSON output from v.db.select or
+                # proper CSV output and real CSV reader here (Python csv and json packages).
+                raise GException(
+                    _(
+                        "Unable to read the table <{table}> from the database due"
+                        " to seemingly inconsistent number of columns in the data transfer."
+                        " Check row: {row}..."
+                        " Likely, a newline character is present in the attribute value starting with: '{value}'"
```

Comment:
Small note: According my testing, existence of newline char in the column value doesn't cause this exception. It does with special separator `{_sep_}`. I added column detection and printing  with values containing this separator, [patch.zip](https://github.com/OSGeo/grass/files/5586105/patch.zip).

Reply:
Thanks for testing this.

I would leave your modifications for another PR. (I'm sorry, in case I misunderstood if you are actually making a suggestion on redoing this PR or not!)

As for newline vs `{_sep_}`, the issue I had was really a newline. Maybe it depends on which newline character it is (LF, CRLF, ...) possibly in combination with platform (I would say, not worth investigating given that better fix of the code is needed anyway, but I can send you the file privately). Given I got the issue with newline, your patch as is would not work in my case, but I share your concern for not misleading the user with the message.

I agree that it would make sense to inform user about the separator in the data, but is it worth all the code needed for that? (and I'm saying that as proponent of helping the user to investigate what happen). The point of `{_sep_}` was that it is strange enough not occur in real data (previous one was something like `|` which is significantly more likely). Rather than increasing complexity, I would just make the separator more strange if `{_sep_}` seems like too likely to occur naturally. Given this, I'm suggesting finish merge this PR as is and discuss this in a separate PR/issue. 

Whether it is newline, separator, or both, I think the TODO, I put there, holds. It is naive parsing of CSV. Either more robust one is needed or JSON. There are Python packages for both. There might have a reason why it was done by hand, but chances are it does not apply anymore.

Finally, you can actually open PRs against branches in other people's forks, well, if that sounds better than sending files to you. I can see advantages in both approaches.

Reply:
> Thanks for testing this.
> 
> I would leave your modifications for another PR. (I'm sorry, in case I misunderstood if you are actually making a suggestion on redoing this PR or not!)
> 
> As for newline vs `{_sep_}`, the issue I had was really a newline. Maybe it depends on which newline character it is (LF, CRLF, ...) possibly in combination with platform (I would say, not worth investigating given that better fix of the code is needed anyway, but I can send you the file privately). Given I got the issue with newline, your patch as is would not work in my case, but I share your concern for not misleading the user with the message.

Yes of course. It's possible that I missed something (during debugging/testing) and it's interesting  because I can't reproduce this exception with newline char `\r\n` or `\n` (condition `if len(columns) != len(record):` will not be met) . 
Because, according debugging this condition can only occur if column value contain this special separator `{_sep_}`.
But I don't know what data you're working with. And that was  reason why I posted my patch and suggested a change.

> I agree that it would make sense to inform user about the separator in the data, but is it worth all the code needed for that? 

I agree that it is better to have a more complex separator.

>Given this, I'm suggesting finish merge this PR as is and discuss this in a separate PR/issue.

Ok.

> Finally, you can actually open PRs against branches in other people's forks, well, if that sounds better than sending files to you. I can see advantages in both approaches.

Ok.